data/base/sub-endpoint.php

```diff
@@ -13,27 +11,89 @@ abstract class SubEndpoint extends Endpoint {
 	/**
 	 * @var string
 	 */
-	protected $parent_route = '';
+	protected $route = '/';
 
-	public function __construct( $parent_route, $parent_endpoint ) {
+	public function __construct( $parent_endpoint, $route = '/' ) {
+		$this->route = $this->ensure_slashes( $route );
 		$this->parent_endpoint = $parent_endpoint;
-		$this->parent_route = $parent_route;
 
 		parent::__construct( $this->parent_endpoint->controller );
 	}
 
 	/**
-	 * Get parent route.
+	 * Ensure start-with and end-with slashes.
+	 *
+	 * '/' => '/'
+	 * 'abc' => '/abc/'
+	 * '/abc' => '/abc/'
+	 * 'abc/' => '/abc/'
+	 * '/abc/' => '/abc/'
+	 *
+	 * @param string $route
+	 *
+	 * @return string
+	 */
+	private function ensure_slashes( $route ) {
+		if ( '/' !== $route[0] ) {
```

Comment:
Use `trailingslashit` and same method for the first slash.

Reply:
Handled: https://github.com/elementor/elementor/pull/12820/commits/57310f6150c9f03cd50ce1c4c27799960a1a2088